src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/CpxVariantDetector.java

```diff
@@ -0,0 +1,782 @@
+        for (final Tuple2<SimpleInterval, Integer> pair : reverse){
+            final int index = pair._2;
+            if (index > 0) {
+                descriptions.add(pair._2, pair._1.toString());
+            } else {
+                descriptions.add(-1*pair._2, "-"+pair._1.toString());
+            }
+        }
+        return descriptions;
+    }
+
+    // avoid case where segment and alignment ref span overlap only on one boundary base
+    private static boolean alignmentContainsSegment(final SimpleInterval alignmentRefSpan, final SimpleInterval segment) {
+        return alignmentRefSpan.overlaps(segment)
+                &&
+                alignmentRefSpan.intersect(segment).size() == segment.size();
+    }
+
+    private static boolean alignmentIsDisjointFromAlphaOmega(final SimpleInterval alignmentRefSpan,
+                                                             final SimpleInterval regionBoundedByAlphaAndOmega) {
+        return !alignmentRefSpan.overlaps(regionBoundedByAlphaAndOmega);
+    }
+
+    // =================================================================================================================
+
+    private static byte[] extractAltHaplotypeSeq(final AssemblyContigWithFineTunedAlignments tigWithInsMappings,
+                                                 final List<SimpleInterval> segments,
+                                                 final BasicInfo basicInfo) {
+
+
+        final List<AlignmentInterval> alignments = tigWithInsMappings.contig.alignmentIntervals;
+        final AlignmentInterval head = alignments.get(0);
+        final AlignmentInterval tail = alignments.get(alignments.size() - 1);
+
+        if (segments.isEmpty()) { // case where middle alignments all map to disjoint locations
+            final int start = head.endInAssembledContig;
+            final int end = tail.startInAssembledContig;
+            final byte[] altSeq = Arrays.copyOfRange(tigWithInsMappings.contig.contigSequence, start - 1, end);
+            if ( ! basicInfo.forwardStrandRep ) {
+                SequenceUtil.reverseComplement(altSeq);
+            }
+            return altSeq;
+        }
+
+        final SimpleInterval firstSegment, lastSegment;
+        if ( basicInfo.forwardStrandRep ) {
+            firstSegment = segments.get(0);
+            lastSegment = segments.get(segments.size() - 1);
+        } else {
+            firstSegment = segments.get(segments.size() - 1);
+            lastSegment = segments.get(0);
+        }
+
+        final int start, end;
+
+        if ( !firstSegment.overlaps(head.referenceSpan) ) {
+            final boolean expectedCase = basicInfo.forwardStrandRep ? (firstSegment.getStart() - head.referenceSpan.getEnd() == 1)
+                                                                    : (head.referenceSpan.getStart() - firstSegment.getEnd() == 1);
+            if ( ! expectedCase )
+                throw new UnhandledCaseSeen(tigWithInsMappings.toString());
```

Comment:
Again please provide a message explaining what's wrong with this situation.

Reply:
expanded and documented